### DISCUSSION TITLE
Slint v1.5 Android: Using `.copy()` and other clipboard methods on `LineEdit` component.

### DISCUSSION THREAD

**Discussion by CM-IV** [slint-ui/slint#4864] [Q&A] [answered]
I have a password generation Slint app that I use in GNU/Linux that has three buttons, one for generating the password via the Rust backend, the second button for copying the password text to clipboard via the `.copy()` function for a read-only `LineEdit` component, and the third button just sets the password as an empty string - clearing the `LineEdit`.

This doesn't work on Android which makes sense as a completely different backend is being used via `slint::android`.

How can I get this working if I want to copy the text of a LineEdit component on Android - via `android_activity` and `AndroidApp`?

**Comment by ogoffart** [comment, marked as answer]
This is not yet implemented yes. I've filled issue https://github.com/slint-ui/slint/issues/4865

We would need to implement the trait method `clipboard_text` and `set_clipboard_text` in 
https://github.com/slint-ui/slint/blob/3aafce2c52fa1e14971e4f2455bdc556daeb079d/internal/backends/android-activity/lib.rs#L89
and call into java (in SlintAndroidJavaHelper.java) to get the clipboard with something like this: https://developer.android.com/reference/android/content/ClipboardManager

**Comment by Frank-schaefer** [comment]
Does this works on Windows ? And if yes , how ? Looking to get the "paste()" working on LineEdit. Thanks

**Reply by ogoffart** [reply]
Yes, this should work on windows.
I guess the best is to continue the discussion in https://github.com/slint-ui/slint/discussions/4910
